### DISCUSSION TITLE
Measure execution time of IKE phases

### DISCUSSION THREAD

**Discussion by yamus-didine** [strongswan/strongswan#1809] [Q&A]
Hello.
I want to measure the execution time of IKE_SA_INIT, CREATE_CHILD_SA
Is there a way to do it from strongswan?
I am currently using wireshark by reading the time of arrival of  each packet.

**Comment by tobiasbrunner** [comment]
What do you mean with execution time exactly? What do you want to measure? And why?

**Reply by yamus-didine** [reply]
Hello.
Probably, I did'nt explain my inquiry well.
So, as a part of my project, I am trying PQC KEM kyber with IPsec (using Liboqs library).
To compare the results I obtained, I am using a recently published paper a reference.
The performance evaluation made  in the paper was based on some metrics, including:
-KEM operations speed (key generation,key encapsulation/decapsulation). 
-Packet size. 
-Latency of some exchanges (`IKE_SA_INIT`, `CREATE_CHILD_SA` and Intermediate key exchanges).

The paper mentionned that the latency of the exchanges is measured by observing the processing time with the task manager of IKEv2 in strongswan deamon. 

I was hoping i could get some guidance.
Thanks in advance.

**Reply by tobiasbrunner** [reply]
What paper is that? With task manager they probably refer to [`task_manager_v2_t`](https://github.com/strongswan/strongswan/blob/master/src/libcharon/sa/ikev2/task_manager_v2.c). They might have patched that to get whatever numbers they wanted.

**Reply by yamus-didine** [reply]
Here is a link to the paper:
https://link.springer.com/chapter/10.1007/978-3-031-29371-9_13

The authors attached a table that shows the evaluation results based on the metrics mentioned previously in this thread.
